test/test_meta.py

```diff
@@ -929,22 +947,86 @@ class MetaCrossRefDispatchMode(torch.utils._python_dispatch.TorchDispatchMode):
     test_case: TestCase
     device: torch.device
     dtype: torch.dtype
+    aten_olp_no_out_overload: set = set()
 
-    def __init__(self, test_case, *, device, dtype, symbolic_meta: bool):
+    def __init__(self, test_case, *, device, dtype, symbolic_meta: bool, inplace: bool, supports_out: bool):
         self.test_case = test_case
         # save TLS
         self.precision = test_case.precision
         self.rel_tol = test_case.rel_tol
         self.device_type = torch.device(device).type
         self.dtype = dtype
         self.symbolic_meta = symbolic_meta
+        self.inplace = inplace
+        self.supports_out = supports_out
 
-    def __torch_dispatch__(self, func, types, args=(), kwargs=None):
-        kwargs = kwargs or {}
+    @staticmethod
+    def try_resolve_aten_out_overload(ol, args, kwargs, num_outputs):
 
-        self.test_case.precision = self.precision
-        self.test_case.rel_tol = self.rel_tol
+        ol_args = ol._schema.arguments
+        olp: OpOverloadPacket = ol._overloadpacket
+
+        if olp in MetaCrossRefDispatchMode.aten_olp_no_out_overload:
+            return (None, None, None)
+
+        candidate_ols = []
+        for candidate_ol_name in olp.overloads():
+            candidate_ol = getattr(olp, candidate_ol_name)
+            if any(arg.is_out for arg in candidate_ol._schema.arguments):
+                candidate_ols.append(candidate_ol)
+
+        if not candidate_ols:
+            MetaCrossRefDispatchMode.aten_olp_no_out_overload.add(olp)
```

Comment:
why adding this before returning?

Reply:
This avoids trying to find an out overload for an `OpOverloadPacket` that is known not to have one from a previous search (Line 969 will return early)